src/main/java/org/mockbukkit/mockbukkit/tags/BaseTagMock.java

```diff
@@ -0,0 +1,51 @@
+package org.mockbukkit.mockbukkit.tags;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Lists;
+import org.bukkit.Keyed;
+import org.bukkit.NamespacedKey;
+import org.bukkit.Tag;
+import org.jetbrains.annotations.NotNull;
+
+public abstract class BaseTagMock<T extends Keyed> implements Tag<T>
```

Comment:
Is extending this class necessary?

The child classes of this class does not provide any behavior that is different from this class.

Reply:
From the class perspective that's true, but we also have the "factory method" in each class to keep things segregated.